packages/node_modules/@node-red/editor-client/src/js/ui/contextMenu.js

```diff
@@ -66,12 +69,13 @@ RED.contextMenu = (function () {
                             })
                         }
                     },
-                    (hasSelection || hasLinks) ? {
+                    (hasSelection || hasLinks) ? { // has nodes or at least 1 wire selected
                         label: RED._("contextMenu.junction"),
                         onselect: 'core:split-wires-with-junctions',
                         disabled: !hasLinks
                     } : {
                         label: RED._("contextMenu.junction"),
+                        disabled: !noSelection, // has 0 nodes, 0 links selected (i.e. workspace right click)
```

Comment:
This change doesn't appear to be related to the original issue. It is valid to add a bare junction when nothing is selected.
```suggestion
```

Reply:
Sorry, the triple inverse logic and comment doesnt help but this has an important (and related function).  It prevents the option "insert→junction" when clicking a single selected virtual wire.

Reply:
Sometimes a written explanation is easier than watching gifs. I know what the desired behaviour is - I'm more concerned with whether the logic is right.

I can see from testing it that it does behave as desired - it was the incorrect comment that threw me off here.

Whilst checking it, I did find one scenario where the option is unexpectedly disabled - if you have some nodes selected (but no wires) the 'insert junction' option is currently disabled - I don't think it should be. We don't disable the 'insert node' option in that scenario. Are you able to take a look at that?

Reply:
> if you have some nodes selected (but no wires) the 'insert junction' option is currently disabled  - I don't think it should be. We don't disable the 'insert node' option in that scenario

i was actually gonna suggest we do not allow insert→node when 1 or more nodes are selected (it doesnt feel right to me 🤷‍♂️ )

> Are you able to take a look at that?

NP, i can make junction clickable when nodes + 0 wires selected.  
Will push change shortly.

Reply:
Ah - now I remember.

I did NOT permit adding junction when 0 wires in the selection because it would mean modifying the code in `function addJunctionsToWires()` - the function expects selection to have actual wires in the selection 

So while I can enable the menu, it wont do anything without further modifications

Reply:
Please advise

Reply:
The 'insert junction' option is represented by two different options - and the code has some logic that picks which one to insert into the menu based on the current selection state. It either invokes `'core:split-wires-with-junctions'` or one that adds a bare junction.

But I can see the logic around which is inserted into the menu isn't quite right to do what I've described.

What I think it should be is:

 - if any (non-virtual) links are selected -> `core:split-wires-with-junctions`
 - if no-links are selected (of any type) -> add junction

With the current logic, it is showing the `core:split-wires-with-junctions` option if there's anything selected but then disabling it if there are no links selected (line 72).

Reply:
ok, that makes sense & I have pushed change to permit add junction as per your idea

Still think its odd :D 

![chrome_xkbFqnKcKR](https://user-images.githubusercontent.com/44235289/178840584-abe9c262-ce18-4dd7-9122-a6e724008431.gif)